core/src/crds.rs

```diff
@@ -209,12 +239,20 @@ impl Crds {
     pub fn remove(&mut self, key: &CrdsValueLabel) -> Option<VersionedCrdsValue> {
         let (index, _, value) = self.table.swap_remove_full(key)?;
         assert!(self.shards.remove(index, &value));
+        if let CrdsData::ContactInfo(_) = value.value.data {
+            assert!(self.nodes.swap_remove(&index));
+        }
         // The previously last element in the table is now moved to the
-        // 'index' position. Shards need to be updated accordingly.
-        if index < self.table.len() {
+        // 'index' position. Shards and nodes need to be updated accordingly.
+        let size = self.table.len();
+        if index < size {
```

Comment:
The only time `index == size`  is if the key being removed is the "last"  element in the `table`. In those cases, the below `insert` is unnecessary, but isn't it still necessary to call the `remove` and `swap_remove` logic below?